include/exadg/operators/variable_coefficients.h

```diff
@@ -23,144 +23,173 @@
 
 namespace ExaDG
 {
-template<int dim, typename Number>
+template<int dim, typename Number, int rank = 0>
 class VariableCoefficientsCells
 {
 private:
-  typedef dealii::VectorizedArray<Number> scalar;
+  using coefficient_t = dealii::Tensor<rank, dim, dealii::VectorizedArray<Number>>;
```

Comment:
I would make this whole expression the template argument of this class. I.e.,
```cpp
template<typename coefficient_t>
class VariableCoefficientsCells;
```

Reply:
Yes, makes a whole lot sense, why not 😄 . I'll see to it.

Reply:
This is done now, and `coefficient_t` is now a template parameter alongside `dim` and `Number`. Could have actually removed `dim` and `Number` as you suggested, however this would not work anymore as we reverted the interface to be initialized with a `MatrixFree<dim, Number>` object again.